### PR TITLE
feat(plugin): connect field style code

### PR DESCRIPTION
**Pull request by airslice**
## Overview

This PR connects filed component style code with the functionality.

## Test

Test project [here](https://test.reearth.dev/edit/01gsydy46k4rvnypx3w2wdf84a).
Test shared project [here](https://test.reearth.dev/published.html?alias=dbgbhdadbc&projectID=01gsyebxnk6r0413gzgbtdn3nn).

## Screen Shot

test reearth dev_edit_01gsydy46k4rvnypx3w2wdf84a(1920 desktop)
test reearth dev_published html_alias=dbgbhdadbc projectID=01gsyebxnk6r0413gzgbtdn3nn(1920 desktop) (1)

## Memo
- Currently apply style with incorrect syntax will easily break the app. Will be fixed on reearth in the future.

### CODE REVIEW

File: plugin/src/sidebar.ts [hunk 1]
```diff
@@ -353,6 +353,16 @@ reearth.on("message", ({ action, payload }: PostMessageProps) => {
     }
   }
 
+  // ************************************************
+  // Style code
+  else if (action === "styleCodeUpdate") {
+    const { dataID, styleObject } = payload;
+    const layerId = addedDatasets.find(ad => ad[0] === dataID)?.[2];
```
Comment (KaWaite) [inline plugin/src/sidebar.ts:360]: layerID

File: plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/general/StyleCode/index.tsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+`;
+
+// const Field = styled.div<{ gap?: number }>`
+//   display: flex;
+//   align-items: center;
+//   ${({ gap }) => gap && `gap: ${gap}px;`}
+//   height: 160px;
+//   padding: 8px;
+//   gap: 8px;
+// `;
```
Comment (KaWaite) [inline plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/general/StyleCode/index.tsx:39]: Please remove unused code

File: plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/types.ts [hunk 3]
```diff
@@ -8,7 +8,7 @@ export const fieldName = {
   timeline: "タイムラインデータ",
   switchGroup: "スイッチグループ",
   buttonLink: "リンクボタン",
-  styleCode: "Style code (General)",
+  styleCode: "Style code",
```
Comment (KaWaite) [inline plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/types.ts:11]: スタイルコード

File: plugin/src/sidebar.ts [hunk 4]
```diff
@@ -353,6 +353,16 @@ reearth.on("message", ({ action, payload }: PostMessageProps) => {
     }
   }
 
+  // ************************************************
+  // Style code
+  else if (action === "styleCodeUpdate") {
```
Comment (KaWaite) [inline plugin/src/sidebar.ts:358]: I don't think you need this route. The "updateDatasetInScene" action is appropriate, right?

File: plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/general/StyleCode/hooks.ts [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+  const [code, editCode] = useState(value.src);
+
+  const updateStyle = useCallback(
+    (styleStr: string) => {
+      if (styleStr) {
+        try {
+          const styleObject = JSON5.parse(styleStr);
+          postMsg({ action: "styleCodeUpdate", payload: { dataID, styleObject } });
+          // eslint-disable-next-line no-empty
+        } catch (error) {}
```
Comment (KaWaite) [inline plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/general/StyleCode/hooks.ts:21]: This try/catch is not in an async function. Don't try/catches need to be in an async func?

File: plugin/src/sidebar.ts (hunk 1 above)
Comment (airslice) [reply plugin/src/sidebar.ts:360]: Ahhh.. updated. I guess i also use `layerId` in other components...

File: plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/general/StyleCode/index.tsx (hunk 2 above)
Comment (airslice) [reply plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/general/StyleCode/index.tsx:39]: Sorry missed this...

File: plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/types.ts (hunk 3 above)
Comment (airslice) [reply plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/types.ts:11]: Updated.

File: plugin/src/sidebar.ts (hunk 4 above)
Comment (airslice) [reply plugin/src/sidebar.ts:358]: Ah yes i can use that. 
And i found it get the layerID twice in "updateDatasetInScene" and removed one.

File: plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/general/StyleCode/hooks.ts (hunk 5 above)
Comment (airslice) [reply plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/general/StyleCode/hooks.ts]: Does try/catch need to be in an async func? 😱

Comment (KaWaite) [reply plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/general/StyleCode/hooks.ts]: Huh....apparently not. Why did I think they needed to be in async funcs.... stupid lol
Anyways, no they don't I guess :P

File: plugin/src/sidebar.ts (hunk 4 above)
Comment (KaWaite) [reply plugin/src/sidebar.ts:358]: You seem to have remoed the double layerId thing but not removed this route. Please remove this.

**Pull request merged**